### PR TITLE
Feature/armory screens

### PR DESCRIPTION
**Pull request by ivandukic**

### CODE REVIEW

File: services/rule_engine/src/main/java/agents/Ava.java [hunk 1]
```diff
@@ -4,13 +4,51 @@
...
+
+import java.util.Map;
+
+import com.mindsmiths.armory.ArmoryAPI;
+import com.mindsmiths.armory.components.DescriptionComponent;
+import com.mindsmiths.armory.components.PrimarySubmitButtonComponent;
+import com.mindsmiths.armory.components.TitleComponent;
+import com.mindsmiths.armory.templates.BaseTemplate;
+import com.mindsmiths.armory.templates.TemplateGenerator;
+
```
Comment (lbenjak) [inline services/rule_engine/src/main/java/agents/Ava.java:23]: extra new line here

File: services/rule_engine/src/main/java/agents/Ava.java [hunk 2]
```diff
@@ -1,25 +1,149 @@
...
+    public void showFamiliarityQuizScreens(){
+        Map<String, BaseTemplate> screens = new HashMap<String, BaseTemplate>();
+        Option[] buttons = Mitems.getOptions("onboarding.familiarity-quiz.buttons");
+        
+        //Adding intro screen
+        String introScreenTitle = Mitems.getText("onboarding.familiarity-quiz.introscreentitle");
+        String introScreenDescription = Mitems.getText("onboarding.familiarity-quiz.introscreendescription");
+        screens.put("introScreen", new TemplateGenerator()
+            .addComponent("image", new ImageComponent("/public/Ava.png"))
```
Comment (mjuraj) [inline services/rule_engine/src/main/java/agents/Ava.java]: Read image path from mitems

File: services/rule_engine/src/main/java/agents/Ava.java [hunk 3]
```diff
@@ -1,25 +1,149 @@
...
+    public void showScreens(String firstScreenId, Map<String, BaseTemplate> screens) {
+        ArmoryAPI.showScreens(getConnection("armory"), firstScreenId, screens);
+    }
+
+    public void showFamiliarityQuizScreens(){
+        Map<String, BaseTemplate> screens = new HashMap<String, BaseTemplate>();
+        Option[] buttons = Mitems.getOptions("onboarding.familiarity-quiz.buttons");
+        
+        //Adding intro screen
+        String introScreenTitle = Mitems.getText("onboarding.familiarity-quiz.introscreentitle");
```
Comment (mjuraj) [inline services/rule_engine/src/main/java/agents/Ava.java]: In mitems, all names should be human readable. So for example - "Intro screen title"

File: services/rule_engine/src/main/java/agents/Ava.java [hunk 4]
```diff
@@ -1,25 +1,149 @@
...
+        ArmoryAPI.showScreen(getConnection("armory"), screen);
+    }
+
+    public void showScreens(String firstScreenId, Map<String, BaseTemplate> screens) {
+        ArmoryAPI.showScreens(getConnection("armory"), firstScreenId, screens);
+    }
+
+    public void showFamiliarityQuizScreens(){
+        Map<String, BaseTemplate> screens = new HashMap<String, BaseTemplate>();
+        Option[] buttons = Mitems.getOptions("onboarding.familiarity-quiz.buttons");
```
Comment (mjuraj) [inline services/rule_engine/src/main/java/agents/Ava.java]: Split into 3 lists

File: services/rule_engine/src/main/java/agents/Ava.java [hunk 5]
```diff
@@ -1,25 +1,149 @@
...
+        int questionNum = 1;
+        while(true){
+            String questionTag = "question" + String.valueOf(questionNum);
+            String nextQuestionTag = "question" + String.valueOf(questionNum + 1);
+
+            try{
+                String questionText = Mitems.getText("onboarding.familiarity-quiz."+ questionTag);
+                screens.put(questionTag, new TemplateGenerator(questionTag)
+                    .addComponent("question", new TitleComponent(questionText))
+                    .addComponent("answers", new CloudSelectComponent("answers", Map.of(
```
Comment (mjuraj) [inline services/rule_engine/src/main/java/agents/Ava.java]: answers -> answers1

**Review by mjuraj — APPROVED** [review]

**Pull request merged**